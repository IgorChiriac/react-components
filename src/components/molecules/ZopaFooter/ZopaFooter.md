### Summary

Use `<ZopaFooter />` to render Zopa's standard footer in a **Zopa application**.

### Examples

- Standard

```jsx
import { ZopaFooter } from '@zopauk/react-components';

<ZopaFooter />;
```

- Only with legal section

```jsx
import { ZopaFooter } from '@zopauk/react-components';

<ZopaFooter legalOnly />;
```
