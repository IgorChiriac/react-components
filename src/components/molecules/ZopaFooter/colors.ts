export const neutral900 = '#1C2139';
export const neutral200 = '#A9ACBA';
export const neutral400 = '#6F748B';
export const neutral800 = '#282F52';
