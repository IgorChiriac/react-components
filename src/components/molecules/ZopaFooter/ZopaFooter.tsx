import React from 'react';
import styled from 'styled-components';

import * as colors from '../../../constants/colors';
import FlexContainer from '../../layout/FlexContainer/FlexContainer';
import Legal from './Legal/Legal';
import Links from './Links/Links';
import MiscLinks from './MiscLinks/MiscLinks';
import Separator from './Separator/Separator';
import SocialLinks from './SocialLinks/SocialLinks';
import Wrapper from './Wrapper/Wrapper';

const StyledWrapper = styled.div`
  background-color: ${colors.neutral.neutral900};
  padding: 40px 0;
`;

export interface IFooterProps {
  /**
   * This renders the legal copy only without any links.
   * This is useful for pages where by design requires the user to focus on one task.
   * @default false
   */
  legalOnly?: boolean;
}

const ZopaFooter: React.FunctionComponent<IFooterProps> = ({ legalOnly }) => (
  <StyledWrapper id="zopa-footer">
    <FlexContainer gutter={16}>
      {legalOnly || <Links />}
      <Legal />
      {legalOnly || (
        <>
          <Wrapper>
            <SocialLinks />
          </Wrapper>
          <Separator />
          <Wrapper>
            <MiscLinks />
          </Wrapper>
        </>
      )}
    </FlexContainer>
  </StyledWrapper>
);

ZopaFooter.defaultProps = {
  legalOnly: false,
};

export default ZopaFooter;
