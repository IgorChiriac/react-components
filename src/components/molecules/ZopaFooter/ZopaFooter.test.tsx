import { axe } from 'jest-axe';
import React from 'react';
import { render } from '@testing-library/react';
import ZopaFooter from './ZopaFooter';
import { mockDate, IMockDate } from '../../../helpers/test/date';

const fixedDate = new Date('2025-02-28T09:39:59');

describe('<ZopaFooter />', () => {
  let date: IMockDate;

  beforeEach(() => {
    date = mockDate(new Date(fixedDate));
    date.startMocking();
  });
  afterEach(() => {
    date.finishMocking();
  });

  it('renders the component with default props', async () => {
    const { container } = render(<ZopaFooter />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the component with a legalOnly prop', async () => {
    const { container } = render(<ZopaFooter legalOnly />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the component with no a11y violations', async () => {
    const { container } = render(<ZopaFooter />);
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
