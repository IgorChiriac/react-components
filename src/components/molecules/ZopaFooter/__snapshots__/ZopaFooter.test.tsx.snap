// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ZopaFooter /> renders the component with a legalOnly prop 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  padding-right: 16px;
  padding-left: 16px;
  margin-left: auto;
  margin-right: auto;
}

.c2 {
  padding: 18px 0;
}

.c3 {
  color: #6F748B;
  font-size: 11px;
  line-height: 1.5;
  margin: 0;
}

.c0 {
  background-color: #1C2139;
  padding: 40px 0;
}

@media (min-width:576px) {
  .c1 {
    max-width: 540px;
  }
}

@media (min-width:768px) {
  .c1 {
    max-width: 720px;
  }
}

@media (min-width:992px) {
  .c1 {
    max-width: 960px;
  }
}

@media (min-width:1200px) {
  .c1 {
    max-width: 1140px;
  }
}

<div
  class="c0"
  id="zopa-footer"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        Zopa Limited is authorised and regulated by the Financial Conduct Authority, and entered on the Financial Services Register (718925). Zopa Bank Limited is authorised by the Prudential Regulation Authority and regulated by the Financial Conduct Authority and the Prudential Regulation Authority, and entered on the Financial Services Register (800542). Zopa Limited (05197592) and Zopa Bank Limited (10627575) are both incorporated in England & Wales and have their registered office at: 1st Floor, Cottons Centre, Tooley Street, London, SE1 2QG.
      </p>
      <br />
      <p
        class="c3"
      >
        © Zopa Bank Limited 2025 All rights reserved. 'Zopa' is a trademark of Zopa Bank Limited.
      </p>
      <br />
      <p
        class="c3"
      >
        Zopa is a member of Cifas – the UK’s leading anti-fraud association, and we are registered with the Office of the Information Commissioner (ZA275984, Z8797078).

    
      </p>
    </div>
  </div>
</div>
`;

exports[`<ZopaFooter /> renders the component with default props 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  padding-right: 16px;
  padding-left: 16px;
  margin-left: auto;
  margin-right: auto;
}

.c2 {
  padding: 18px 0;
}

.c12 {
  color: #6F748B;
  font-size: 11px;
  line-height: 1.5;
  margin: 0;
}

.c4 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: 16px;
  padding-left: 16px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

.c6 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: 16px;
  padding-left: 16px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

.c10 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: px;
  padding-left: px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-right: -16px;
  margin-left: -16px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-weight: 600;
  line-height: 1.4;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 16px;
  color: #A9ACBA;
  -webkit-transition: color 0.2s ease;
  transition: color 0.2s ease;
}

.c5:hover {
  color: #6F748B;
}

.c11 {
  background-color: #282F52;
  height: 1px;
}

.c7 {
  font-size: 18px;
  color: #fff;
  font-weight: 600;
  line-height: 24px;
  margin-bottom: 16px;
}

.c8 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.c9 {
  margin: 0 0 8px 0;
  padding: 0;
}

.c0 {
  background-color: #1C2139;
  padding: 40px 0;
}

@media (min-width:576px) {
  .c1 {
    max-width: 540px;
  }
}

@media (min-width:768px) {
  .c1 {
    max-width: 720px;
  }
}

@media (min-width:992px) {
  .c1 {
    max-width: 960px;
  }
}

@media (min-width:1200px) {
  .c1 {
    max-width: 1140px;
  }
}

@media (min-width:0px) {
  .c4 {
    display: block;
    -webkit-flex: 0 0 auto;
    -ms-flex: 0 0 auto;
    flex: 0 0 auto;
    width: auto;
    max-width: none;
  }
}

@media (min-width:0px) {
  .c6 {
    display: block;
    -webkit-flex: 0 0 100%;
    -ms-flex: 0 0 100%;
    flex: 0 0 100%;
    max-width: 100%;
  }
}

@media (min-width:576px) {
  .c6 {
    display: block;
    -webkit-flex: 0 0 50%;
    -ms-flex: 0 0 50%;
    flex: 0 0 50%;
    max-width: 50%;
  }
}

@media (min-width:768px) {
  .c6 {
    display: block;
    -webkit-flex: 0 0 33.33333333333333%;
    -ms-flex: 0 0 33.33333333333333%;
    flex: 0 0 33.33333333333333%;
    max-width: 33.33333333333333%;
  }
}

@media (min-width:992px) {
  .c6 {
    display: block;
    -webkit-flex: 0 0 25%;
    -ms-flex: 0 0 25%;
    flex: 0 0 25%;
    max-width: 25%;
  }
}

@media (min-width:0px) {
  .c10 {
    display: block;
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:576px) {
  .c10 {
    display: none;
  }
}

<div
  class="c0"
  id="zopa-footer"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        cols="12"
        direction="row"
      >
        <div
          class="c4"
          cols="12"
        >
          <a
            aria-label="twitter"
            class="c5"
            href="https://twitter.com/zopa"
          >
            <svg
              fill="currentColor"
              height="30px"
              role="img"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="30px"
            >
              <path
                d="M19 0H5a5 5 0 0 0-5 5v14a5 5 0 0 0 5 5h14a5 5 0 0 0 5-5V5a5 5 0 0 0-5-5zm-.139 9.237c.209 4.617-3.234 9.765-9.33 9.765a9.286 9.286 0 0 1-5.032-1.475 6.605 6.605 0 0 0 4.86-1.359 3.285 3.285 0 0 1-3.066-2.28 3.3 3.3 0 0 0 1.482-.056c-1.579-.317-2.668-1.739-2.633-3.26.442.246.949.394 1.486.411A3.289 3.289 0 0 1 5.612 6.6a9.32 9.32 0 0 0 6.766 3.43 3.286 3.286 0 0 1 5.594-2.993 6.583 6.583 0 0 0 2.086-.796 3.296 3.296 0 0 1-1.443 1.816A6.578 6.578 0 0 0 20.5 7.54a6.66 6.66 0 0 1-1.639 1.697z"
              />
            </svg>
          </a>
        </div>
        <div
          class="c4"
          cols="12"
        >
          <a
            aria-label="facebook"
            class="c5"
            href="https://facebook.com/zopa"
          >
            <svg
              fill="currentColor"
              height="30px"
              role="img"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="30px"
            >
              <path
                d="M19 0H5a5 5 0 0 0-5 5v14a5 5 0 0 0 5 5h14a5 5 0 0 0 5-5V5a5 5 0 0 0-5-5zm-3 7h-1.924C13.461 7 13 7.252 13 7.889V9h3l-.238 3H13v8h-3v-8H8V9h2V7.077C10 5.055 11.064 4 13.461 4H16v3z"
                fill="currentColor"
                fill-rule="nonzero"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
        cols="12"
        direction="row"
      >
        <div
          class="c6"
          cols="12"
        >
          <div
            class="c2"
          >
            <div
              class="c7"
            >
              About Zopa
            </div>
            <ul
              class="c8"
            >
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/about"
                >
                  About Zopa
                </a>
              </li>
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/about/how-zopa-works"
                >
                  How Zopa works
                </a>
              </li>
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/about/join-us"
                >
                  Join us
                </a>
              </li>
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/about/press"
                >
                  Press office
                </a>
              </li>
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/contact"
                >
                  Contact Zopa
                </a>
              </li>
            </ul>
          </div>
          <div
            class="c10"
          >
            <div
              class="c11"
            />
          </div>
        </div>
        <div
          class="c6"
          cols="12"
        >
          <div
            class="c2"
          >
            <div
              class="c7"
            >
              Zopa Loans
            </div>
            <ul
              class="c8"
            >
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/loans/debt-consolidation"
                >
                  Debt consolidation
                </a>
              </li>
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/loans/home-improvement"
                >
                  Home improvements
                </a>
              </li>
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/loans/car-loans"
                >
                  Car loans
                </a>
              </li>
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/loans/faq"
                >
                  FAQs for loans
                </a>
              </li>
            </ul>
          </div>
          <div
            class="c10"
          >
            <div
              class="c11"
            />
          </div>
        </div>
        <div
          class="c6"
          cols="12"
        >
          <div
            class="c2"
          >
            <div
              class="c7"
            >
              Invest your money
            </div>
            <ul
              class="c8"
            >
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/lending"
                >
                  Invest at Zopa
                </a>
              </li>
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/lending/risk-management"
                >
                  Risk management
                </a>
              </li>
              <li
                class="c9"
              >
                <a
                  class="c5"
                  href="https://zopa.com/lending/faq"
                >
                  FAQs for investing
                </a>
              </li>
            </ul>
          </div>
          <div
            class="c10"
          >
            <div
              class="c11"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="c11"
    />
    <div
      class="c2"
    >
      <p
        class="c12"
      >
        Zopa Limited is authorised and regulated by the Financial Conduct Authority, and entered on the Financial Services Register (718925). Zopa Bank Limited is authorised by the Prudential Regulation Authority and regulated by the Financial Conduct Authority and the Prudential Regulation Authority, and entered on the Financial Services Register (800542). Zopa Limited (05197592) and Zopa Bank Limited (10627575) are both incorporated in England & Wales and have their registered office at: 1st Floor, Cottons Centre, Tooley Street, London, SE1 2QG.
      </p>
      <br />
      <p
        class="c12"
      >
        © Zopa Bank Limited 2025 All rights reserved. 'Zopa' is a trademark of Zopa Bank Limited.
      </p>
      <br />
      <p
        class="c12"
      >
        Zopa is a member of Cifas – the UK’s leading anti-fraud association, and we are registered with the Office of the Information Commissioner (ZA275984, Z8797078).

    
      </p>
    </div>
  </div>
</div>
`;
